### ISSUE TITLE
Atomic sites unable to upload Keynote and Zip files

### ISSUE THREAD

**Issue by adelineyaw** [Automattic/wp-calypso#18601]
User has been trying to upload a Keynote file in Calypso but has been unsuccessful. They are currently on a free plan. 

I've tried using both my own test file and the users file in Calypso and wp-admin. The following error messages appear:

https://cloudup.com/cw4RtaZyt2Q (Calypso)

https://cloudup.com/cKBXvMmKrqQ (wp-admin)

The only test site that was successful in uploading a Keynote was on a Business Plan. However an Atomic site and Free plan site did not work.

Our support doc doesn't indicate a specific plan that users need to be on: https://en.support.wordpress.com/accepted-filetypes/

Ticket: 704000-zen

* Fellow HE was able to test the file and it didn't upload on a free plan test site.
* User's file to test can be found in the ticket link above.
* Internal ref: p58i-4Lg-p2

**Comment by alisterscott** [comment]
Apple Keynote files are actually zip files, which is shown by doing the following in a terminal window:

```
$ file -I example.key 
example.key: application/zip; charset=binary
```

So only Business plans can upload these files as only Business plans support uploading .zip files: https://en.support.wordpress.com/accepted-filetypes/

Since we have no way to change what these files actually are, and require a filetype check regardless of extension, I have updated our [support doco](https://en.support.wordpress.com/accepted-filetypes/) to state that .key are only available for sites on plans:

> With the purchase of the WordPress.com Personal, Premium or Business plan, you are also permitted to upload:
> 
> .key (Apple Keynote Presentation)
> .zip (Archive File Format)

I'll close this as the documentation has been updated - a customer on a free plan could be told to export a keynote as a PDF or Powerpoint if they wish to upload a presentation.

**Comment by adelineyaw** [comment]
Just wanted to add that I was unable to upload a keynote file to an Atomic test site via Calypso but I could upload the file via wp-admin. I get the same error message as above:  https://cloudup.com/cw4RtaZyt2Q

**Comment by rachelmcr** [comment]
I tested and confirmed that I can't upload Keynote or ZIP files to Atomic sites. I _also_ can't upload them to self-hosted Jetpack sites, regardless of the host, so this isn't specific to Atomic sites.

In each case, the API returns this error:

`{"code":400,"headers":[{"name":"Content-Type","value":"application\/json"}],"body":{"error":"unsupported_mime_type","message":"File type unknown"}}`

It looks like this error is coming from our Jetpack client, where `$allowed_mimes` does not include `key` or `zip` files.

**Comment by davoraltman** [comment]
I can confirm that I can reproduce this with a .zip file, on both JP and Atomic sites.

This is the error I get - https://cld.wthms.co/roOZ6K

It works when I try uploading from WP Admin.

Chat: 1511665-hc

**Comment by nielslange** [comment]
Another report of a customer being unable to upload a zip file in 7681600-hc

**Comment by katiebethbrown** [comment]
Another report in 17398690-hc

**Comment by JoshuaGoode** [comment]
This is related to https://github.com/Automattic/wp-calypso/issues/30426

In a recent test, I was technically able to upload the .zip file via Calypso; however, it initially reported that the upload failed.  Upon refresh and checking my WP-Admin, it was there.

Here's the error I saw in Calypso:

![image](https://user-images.githubusercontent.com/4887830/109861209-0a1d5880-7c2d-11eb-810f-7a28d80a46d0.png)

I then attempted to upload via WP-Admin directly and the upload would never complete.  It would just stall with this.

![image](https://user-images.githubusercontent.com/4887830/109861269-1ef9ec00-7c2d-11eb-97c3-0ab505cc1f49.png)
